lib/commands/general.js

```diff
@@ -134,9 +134,20 @@ commands.background = async function background (seconds) {
     await this.adb.goToHome();
     return true;
   }
+  const msToWait = seconds * 1000;
   let {appPackage, appActivity} = await this.adb.getFocusedPackageAndActivity();
   await this.adb.goToHome();
-  await B.delay(seconds * 1000);
+
+  // B.delay is not reliable for long periods. So if we have a short background period, just use it
+  // directly, otherwise spin in increments of 1s for as long as we need to.
+  if (seconds < 60) {
```

Comment:
we could use waitForCondition instead from the module you wrote ;)

Reply:
LOL true, the ergonomics of that aren't great. Let me do something else.